runtime/compiler/z/codegen/S390J9CallSnippet.cpp

```diff
@@ -515,12 +518,17 @@ TR::S390VirtualUnresolvedSnippet::emitSnippetBody()
                              __FILE__, __LINE__, callNode);
    cursor += sizeof(uintptrj_t);
 
+   // CP addr
    *(uintptrj_t *) cursor = (uintptrj_t) callNode->getSymbolReference()->getOwningMethod(comp)->constantPool();
 
-   AOTcgDiag1(comp, "add TR_Thunks cursor=%x\n", cursor);
-   cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor, *(uint8_t **)cursor, getNode() ? (uint8_t *)(intptr_t)getNode()->getInlinedSiteIndex() : (uint8_t *)-1, TR_Thunks, cg()),
-                             __FILE__, __LINE__, callNode);
+   AOTcgDiag1(comp, "add TR_J2IVirtualThunkPointer cursor=%x\n", cursor);
+   cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor, (uint8_t*)j2iRelocInfo, NULL,
+                                                                                 TR_J2IVirtualThunkPointer, cg()),
```

Comment:
I'd prefer if we initialize `j2iRelocInfo` before passing it as an argument here. What if `TR::ExternalRelocation` were to cache some of the results from this argument? We would be reading garbage memory then. This relocation should be moved to after `data3` is initialized.

Reply:
The `j2iRelocInfo` is now fully initialized before used.